fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/ComparisonRange.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * ComparisonRange.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2019 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.plan.temp;
+
+import com.apple.foundationdb.API;
+import com.apple.foundationdb.record.RecordCoreException;
+import com.apple.foundationdb.record.query.expressions.Comparisons;
+import com.apple.foundationdb.record.query.plan.ScanComparisons;
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * A set of compatible comparisons on a single field of a{@link com.apple.foundationdb.record.metadata.expressions.KeyExpression}
+ * representing a contiguous range of values for that field.
+ *
+ * <p>
+ * A {@code ComparisonRange} is similar to a {@link ScanComparisons} but represents a contiguous range of values for a
+ * single field, rather than for an entire {@code KeyExpression}. A comparison range is effectively a sum type, with
+ * the following options:
+ * </p>
+ * <ul>
+ *     <li>An empty set of comparisons, indicating that the entire universe of possible values is in the range.</li>
+ *     <li>
+ *         An equality comparison on the field. There can only be a single such comparison. For example, single
+ *         comparison of the form {@code EQUALS "foo"}.
+ *     </li>
+ *     <li>
+ *         A set of inequality comparisons that define a single contiguous range of values. For example, the combination
+ *         of the comparisons {@code > 8} and {@code < 30}. The comparison range may include redundant
+ *         comparisons by contract, but it may or may not simplify the range into a more compact form. For example,
+ *         the comparison range can include the comparisons {@code > 8}, {@code < 30}, and {@code EQUALS 10}, but it
+ *         may optionally simplify this to {@code > 8} and {@code < 30}. Note that this behavior is not fully
```

Comment:
Doesn't this simplify to ~~`EQUALS 10`~~ `> 8` and `< 20`?

Reply:
Whoops, yes. Fixed.